packages/flutter/lib/src/material/switch.dart

```diff
@@ -1078,3 +1322,288 @@ class _SwitchPainter extends ToggleablePainter {
     super.dispose();
   }
 }
+
+class _SwitchDefaults {
```

Comment:
So is the reason for this class because we have a bunch of values that need to change between M2 and M3, but they aren't in the SwitchThemeData? If so, perhaps we should just add them to the ThemeData so that apps can customize them if they want. That would be consistent with the way we handle other components.  Although there is a lot here, that maybe we don't want to bloat the ThemeData with.

If we don't want to make them part of the ThemeData, then it might be cleaner to just have a `_SwitchConfig` interface and have a hand coded `_SwitchConfigM2` and the generated `_SwitchConfigM3` that implement it. That will make it easier to ditch the M2 implementation when we switch over to M3 completely.

Reply:
Thanks a lot for the suggestions! Just created a separate config class for both M2 and M3. Please let me know if there're any problems:)